pyopcode/api.cpp

```diff
@@ -1,122 +1,201 @@
-#include <iostream>
+
+#include <stdexcept>
+
 #include <boost/cstdfloat.hpp>
```

Comment:
Ah, I thought you got rid of boost completely, but no

Reply:
Yeah; using the header-only part of boost isnt so bad though (not the compilation/linking/distribution disaster that the rest of it is). And for some reason the cpp standard does not think its useful to provide people with float types of a guaranteed number of bytes ('it all depends on a million factors involving your compiler version and hardware' used to be a good thing back in the day I suppose), but thats definitely what we want here, and boost provides.

Reply:
I'm mostly concerned about the packaging. Is boost-cpp available on Pypi?

Reply:
Right, the conda-forge recipe gets it from https://dl.bintray.com/boostorg/release/{{ version }}/source/{{ filename }}

Hence Korijn's comment to include it via a submodule

Reply:
Yeah, adding boost via a git submodule is definitely an option. There are probably other ways of getting a guaranteed float32 in cpp, but this is the one I happen to know. But getting rid of the dependency altogether may very well be possible too.

Reply:
I just downloaded the source tarball from the website for now, although a git submodule would be neater.